csrc/python_frontend/fusion_record.h

```diff
@@ -726,48 +726,40 @@ struct SqueezeOpRecord : RecordFunctor {
 };
 
 //! Specialized Record Functor for the FusionState's broadcast_in_dim op.
-
-template <typename OutputShapeType>
+// NOTE: output_size gives the rank of the output tensor.  This size can be
+// found from the State after the definition is read and the Fusion IR is in the
+// process of being created.  However, pior to that point, the size is needed
+// for matching a Fusion Record node in the Trie used to cache definitions.
 struct BroadcastInDimOpRecord : RecordFunctor {
   BroadcastInDimOpRecord(
       std::vector<State> _args,
       std::vector<State> _outputs,
-      std::string _name,
-      serde::RecordType record_type,
-      std::vector<OutputShapeType> output_shape,
+      size_t output_size,
       std::vector<int64_t> broadcast_dims)
       : RecordFunctor(
             std::move(_args),
             std::move(_outputs),
-            _name,
-            record_type),
-        output_shape_(std::move(output_shape)),
-        broadcast_dims_(std::move(broadcast_dims)) {}
+            "ops.broadcast_in_dim",
+            serde::RecordType_BroadcastInDim),
+        output_size_(output_size),
+        broadcast_dims_(std::move(broadcast_dims)) {
+    arg_names_[1] = "shape";
+  }
   ~BroadcastInDimOpRecord() override = default;
   RecordFunctor* clone() final {
     return new BroadcastInDimOpRecord(*this);
   }
 
-  inline size_t outputShapeHash(
-      const std::vector<OutputShapeType>& shape) const;
-
   //! Child specific hash function in lower 32 bits.
-  //! | 31 -------------- 16 | 15 --------------  0 |
-  //! | broadcast_dims hash  | output_shape hash    |
-  //!
-  //! The output_shape hash is specialized in 2 ways using the method
-  //! outputShapeHash:
-  //! 1. int64_t - hashes dimension sizes.
-  //! 2. State - hashes number of dimensions
+  //! | 31 -------------------------------------  0 |
+  //! | broadcast_dims hash                         |
   size_t hash() const final {
     auto result = RecordFunctor::hash();
     size_t broadcast_dims_hash = 0;
     for (auto dim : broadcast_dims_) {
-      broadcast_dims_hash |= 1 << ((output_shape_.size() - 1) - dim);
+      broadcast_dims_hash |= 1 << ((output_size_ - 1) - dim);
```

Comment:
Unimportant: could this just be filled from the right instead?
```suggestion
      broadcast_dims_hash |= 1 << dim;
```